File: 2024/day01/day01.py
with open("input") as fd:
    input = ([int(x) for x in l.split()] for l in fd.readlines())
left, right = map(sorted, zip(*input))
print(sum(abs(l - r) for l, r in zip(left, right)))
print(sum(sum(l for x in right if x == l) for l in left))

File: 2024/day01/day01.ts
#!/usr/bin/env -S deno run --allow-read
const input = await Deno.readTextFile("input");
const parsedInput = input
  .split("\n")
  .map((s) => s.split("   "))
  .filter((l) => l.length == 2)
  .map((l) => [parseInt(l[0]), parseInt(l[1])]);
const left = parsedInput.map((l) => l[0]).sort();
const right = parsedInput.map((l) => l[1]).sort();
console.log(
  left.map((x, i) => Math.abs(x - right[i])).reduce((t, c) => t + c, 0),
);
const counts = Object.groupBy(right, (x) => x);
console.log(
  left.map((n) => n * (counts[n]?.length || 0)).reduce((t, c) => t + c, 0),
);
